app/models/person.rb

```diff
@@ -20,20 +20,22 @@ def attribute_to_entity_property(property)
     case property.to_sym
     when :id
       nil
-    when :client_integration_id
-      return cas_guid if cas_guid.present?
-      return ea_guid if ea_guid.present?
-      return email if email.present?
-      id
     when :authentication
-      { key_guid: cas_guid }
+      { key_guid: cas_guid } if cas_guid.present?
     when :email_address
       [{ email: email, client_integration_id: email }] if email.present?
     else
       super
     end
   end
 
+  def client_integration_id
+    return cas_guid if cas_guid.present?
```

Comment:
Ah, makes sense. Then another option is `[cas_guid, ea_guid, email, id].find(&:present?)`. But I'm not excited about that one-liner. If you like it, then you can use it, otherwise I give this PR a 👍